src/asset-proxy.ts

```diff
@@ -0,0 +1,32 @@
+import { Url, getPlaceholder, isImageResponse } from './utils'
+
+export async function assetProxy(request: Request): Promise<Response | null> {
+  const url = Url.fromRequest(request)
+
+  if (url.subdomain !== 'asset-proxy') return null
+  if (url.pathname !== '/src') return null
+
+  const urlParam = url.searchParams.get('url')
+
+  if (!urlParam) return getPlaceholder()
+
+  let assetUrl: Url
+
+  try {
+    assetUrl = new Url(urlParam)
+  } catch {
+    return getPlaceholder()
+  }
+
+  const originalResponse = await fetch(assetUrl, {
+    cf: { cacheTtl: 24 * 60 * 60 },
+  })
+
+  if (originalResponse.ok && isImageResponse(originalResponse)) {
+    const response = new Response(originalResponse.body, originalResponse)
+    response.headers.delete('set-cookie')
+    return response
+  }
```

Comment:
should we add our own cache header here?
(ok responses should be cached pretty long or even forever I think)

Reply:
You mean, in order that frontend caches the response? 
Otherwise, it is already cached by cloudflare worker. 
Should I increase the cacheTtl? 
Just think about the following case: someone work on an image in real time, that is made available by just one url. Is it realistic?

Reply:
> Just think about the following case: someone work on an image in real time, that is made available by just one url. Is it realistic?

You are right, there are edge cases. For a case like that even one day would be pretty annoying :D 
But considering that the alternative solution is storing them on our server with no updates from the source url and that our power users can just upload the image if they need a new version I'm happy to increase the cache time.

Reply:
So yes, I'd increase the cacheTtl of the fetch. Btw: CF-Workers cache images on fetch by default. I could not find out how long for.

But what setting a cache header here could instruct the users browsers to cache the image.
This would be a pretty agressive cache (year) :)
`response.headers.set("Cache-Control", "public, max-age=31536000, immutable")`

Reply:
Will try to make my feedback more explicit next time.

Reply:
Don't worry, that was great, thanks, I've learned something.

>  could not find out how long for.

I think the information you want can be found [here](https://developers.cloudflare.com/cache/how-to/edge-browser-cache-ttl/#browser-cache-ttl).

I feel unconfortable for asking the user's browser to store anything for so long. But I'll increase it to one month.

Reply:
> I feel unconfortable for asking the user's browser to store anything for so long. But I'll increase it to one month.

That's what CDNs do, it's also the setting on our images for asset.serlo.org/…

Reply:
But one month is also okay for now, we should monitor the usage/cost in any case.

Reply:
> That's what CDNs do, it's also the setting on our images for asset.serlo.org.

you are right, while developing I noticed pexels also does that. I will make it one year.